### ISSUE TITLE
Chain Specification Values Definitions

### ISSUE THREAD

**Issue by stone212** [paritytech/parity#6668]
I think the Parity team does a very good job and I do not wish for this issue to offend the people who put hard work into writing and documenting the code.  But I do find that the documentation is very inadequate regarding the definitions of the values in the Chain Specification file:

https://github.com/paritytech/parity/wiki/Chain-specification

I request that the team *please* take the time to write new, better definitions for these values.  As we all learn in school, a definition of a term should not have the term in it!  One example:

`"gasLimit" gas limit of the genesis, affects the initial gas limit adjustment`

So this definition is saying that if you want to know what "gasLimit" in the genesis file means, you are told it is the gas limit of the genesis. !!#@!  That is like saying that a circle is the quality that all circular things have.  And every definition in that page is similar.  Another example:

`"transactionsRoot" genesis transactions root`

Really?  If I don't know what "transactionsRoot" means then how can it help to know that it is the "genesis transactions root" right?

How about this:

`"minGasLimit" gas limit can adjust across blocks, this parameter determines the absolute minimum it can reach.
`
That is a little better, but the first part tells me that <thing-I-don't-understand> can adjust across blocks (how?), and that <thing-I-don't-understand> determines the minimum value of <thing-I-don't-understand>.  So I ask "what is a gas limit" and I go to the first example. :)

I know that documentation is difficult, and generally you people are doing a great job at coding, documenting, debugging, etc.  This is meant as friendly criticism from someone who is very impressed with how good the Parity team is, and wants to understand Parity better by understanding each of the genesis file terms.

**Comment by rphmeier** [comment]
I'm not sure what the best plain english resource for explaining gas limits or merkle tries as used in Ethereum, but they are comprehensively covered in the yellow paper. 

The gas limit of a block is a dynamic "block size" parameter, which dictates how much computation the transactions within it can do. The chain can adjust to the target gas limit of the miners/sealers, so it doesn't _really_ matter what it starts as in the genesis, but choosing a value that won't take too long to converge to your desired target would be convenient.

The transactions/receipts trie roots aren't important unless you're doing something very fancy. I'd classify these as "only if you really know what you're doing" parameters.

**Comment by stone212** [comment]
> The genesis transactions/receipts trie roots aren't important unless you're doing something very fancy.

> I'd classify these as "only if you really know what you're doing" parameters.

Me too.  That's why I am hoping that the excellent Parity team will improve the documentation.  Then I can learn what the terms mean and play with them until I know what I'm doing.  How else would I get to that level?

> or merkle tries

Who mentioned merkle trees?  Are those even included in a chain spec file in some way?

**Comment by AyushyaChitransh** [comment]
@stone212 There are plans for improving the wiki and issue #6585 takes care of that.

**Comment by rphmeier** [comment]
@stone212 The `*Root` fields you brought up refer to root hashes of merkle tries. these root hashes are contained in the header.

IMO when comparing the YP definition of a block header to the genesis structure in the spec, it is pretty evident which fields map to what. The merkle trie root fields don't really need to be messed with, but the gas limit fields are worth configuring in many cases.

**Comment by stone212** [comment]
@rphmeier 

> The *Root fields you brought up refer to root hashes of merkle tries

I would not have guessed this in years of trying.  This is a perfect example of why I am requesting the team work on the definitions.  Really, who would guess this, except someone who is already an expert.  The purpose of documentation is to go from nothing to expertise.

> IMO when comparing the YP definition of a block header to the genesis structure in the spec, it is pretty evident which fields map to what

Perhaps you are an expert?  Or just smarter than I am?  But IMO, it is not at all clear.  I'm not asking about a TL;DR.  I read everything.  But I would never guess that a word like "root" means "merkle tree" or anything that that is not written fully and using the same language.

**Comment by stone212** [comment]
@AyushyaChitransh Making the wiki community-editable can be very helpful and I can think of many things I would like to update.  But that would not solve this problem.  Because people who know what the chain spec values mean would not think to edit the wiki and people who don't would not be able to.

What is needed is a definitive and thorough explanation of any "jargon".  It has to be written in plain English in a way that someone can learn what the jargon means.  That is what a glossary is for, and since this is a glossary of Parity-specific (though sometimes inheretic) terms, it is necessary for the Parity team to do this.

If someone wants to take the time to explain the terms to me, I will edit the wiki with the explanations I mention.  But I need to know the definitions first, and the only place I could learn that is on the wiki page I would be editing.  Or by chatting with the parity team, but in that case you could just use that time to edit the wiki yourselves. :)

**Comment by AyushyaChitransh** [comment]
@stone212 The wiki is now editable and this issue can now be addressed in https://github.com/paritytech/wiki

**Comment by debris** [comment]
the documentation is available here https://wiki.parity.io/Chain-specification.html

**Comment by stone212** [comment]
@debris This is the same inadequate information that caused me to open the issue in the first place.  Did you read the thread?

@AyushyaChitransh I just now saw that the wiki is editable.  Maybe I can help to create a list of Chain Specification Values Definitions if other people who actually know the values and their definitions want to work with me.  I can add the few that I have surmised.  But I went to https://wiki.parity.io/Chain-specification.html and did not see a way to edit the page.

**Comment by tomusdrw** [comment]
@stone212 Wiki is now in that repository:
https://github.com/paritytech/wiki/

This specific file can be found here:
https://github.com/paritytech/wiki/blob/master/Chain-specification.md

You can submit a Pull Request to alter the file content. Feel free to contact me on Gitter to clarify all the terms.
